compiler/tests-spec/testData/diagnostics/notLinked/dfa/pos/59.fir.kt

```diff
@@ -187,7 +187,7 @@ fun case_14() {
     do {
         <!DEBUG_INFO_EXPRESSION_TYPE("kotlin.Any?")!>x<!>
         x = x<!UNSAFE_CALL!>.<!>equals(10)
-    } while (x is Any)
+    } while (<!USELESS_IS_CHECK!>x is Any<!>)
```

Comment:
This and below seem what FIR DFA is doing better (i.e., FE1.0 didn't warn). With proper checks (`if (x == null) return` or `if (x is Any) { ... }`), `x` starts with non-null `Any` in the beginning of the loop, and then after `x = ... equals ...`, smartcast info for `x` is `Boolean`, hence redundant `is Any`.

It's irony, though, that, with special handling via preliminary loop visitor, `x`'s type before reassignment is still `Any?` so that `UNSAFE_CALL` behavior matches, hmm.